File: backend/src/main/java/com/example/ecommerce/service/ProductService.java
package com.example.ecommerce.service;

import com.example.ecommerce.dto.ProductDTO;
import com.example.ecommerce.dto.request.ProductRequest;
import com.example.ecommerce.dto.request.StockUpdateRequest;

import java.util.List;

public interface ProductService {
    List<ProductDTO> getAllProducts();
    ProductDTO getProductById(Long id);
    List<ProductDTO> getProductsByCategory(Long categoryId);
    List<ProductDTO> searchProducts(String keyword);
    ProductDTO createProduct(ProductRequest request);
    ProductDTO updateProduct(Long id, ProductRequest request);
    void deleteProduct(Long id);
    ProductDTO updateStock(Long id, StockUpdateRequest request);
}

File: frontend/src/services/productService.js
import apiClient from './api'

export const getAllProducts = (params = {}) =>
  apiClient.get('/products', { params }).then((res) => res.data)

export const getProductById = (id) =>
  apiClient.get(`/products/${id}`).then((res) => res.data)

export const searchProducts = (keyword) =>
  apiClient.get('/products', { params: { search: keyword } }).then((res) => res.data)

export const getProductsByCategory = (categoryId) =>
  apiClient.get('/products', { params: { categoryId } }).then((res) => res.data)

export const createProduct = (payload) =>
  apiClient.post('/products', payload).then((res) => res.data)

export const updateProduct = (id, payload) =>
  apiClient.put(`/products/${id}`, payload).then((res) => res.data)

export const deleteProduct = (id) => apiClient.delete(`/products/${id}`)

export const updateProductStock = (id, stock) =>
  apiClient.patch(`/products/${id}/stock`, { stock }).then((res) => res.data)
